Sources/Commands/SwiftPackageCollectionsTool.swift

```diff
@@ -0,0 +1,358 @@
+
+        mutating func run() throws {
+            guard let searchQuery = searchQuery else {
+                throw CollectionsError.missingArgument("searchQuery")
+            }
+
+            switch searchMethod {
+            case .keywords:
+                let results = try await { collections.findPackages(searchQuery, collections: nil, profile: profileOptions.usedProfile, callback: $0) }
+
+                results.items.forEach {
+                    print("\($0.package.repository.url): \($0.package.description ?? "")")
+                }
+
+            case .module:
+                let results = try await { collections.findTargets(searchQuery, searchType: .exactMatch, collections: nil, profile: profileOptions.usedProfile, callback: $0) }
+
+                results.items.forEach {
+                    $0.packages.forEach {
+                        print("\($0.repository.url): \($0.description ?? "")")
+                    }
+                }
+            }
+        }
+    }
+
+    // MARK: Packages
+
+    struct Describe: ParsableCommand {
+        static var configuration = CommandConfiguration(abstract: "Get metadata for a single package")
+
+        @OptionGroup
+        var jsonOptions: JSONOptions
+
+        @OptionGroup
+        var profileOptions: ProfileOptions
+
+        @Argument(help: "URL of the package to get information for")
+        var packageUrl: String?
+
+        @Option(name: .long, help: "Version of the package to get information for")
+        var version: String?
+
+        private func printVersion(_ version: PackageCollectionsModel.Package.Version?) -> String? {
+            guard let version = version else {
+                return nil
+            }
+
+            let modules = version.targets.compactMap { $0.moduleName }.joined(separator: ", ")
+            let platforms = optionalRow("Verified Platforms", version.verifiedPlatforms?.map { $0.name }.joined(separator: ", "))
+            let swiftVersions = optionalRow("Verified Swift Versions", version.verifiedSwiftVersions?.map { $0.rawValue }.joined(separator: ", "))
+            let license = optionalRow("License", version.license?.type.description)
+            let cves = optionalRow("CVEs", version.cves?.map { $0.identifier }.joined(separator: ", "))
+
+            return """
+                \(version.version)
+                Package Name: \(version.packageName)
+                Modules: \(modules)\(platforms)\(swiftVersions)\(license)\(cves)
+            """
+        }
+
+        mutating func run() throws {
+            guard let packageUrl = packageUrl else {
+                throw CollectionsError.missingArgument("packageUrl")
+            }
+
+            let identity = PackageReference.computeIdentity(packageURL: packageUrl)
+            let reference = PackageReference(identity: identity, path: packageUrl)
+
+            let result = try await { self.collections.getPackageMetadata(reference, profile: profileOptions.usedProfile, callback: $0) }
```

Comment:
hmm I think we may have a mistake in the API here: profile does not seem relevant for this call

Reply:
Profile is relevant. A package may show up in more than one collection and possibly with different versions. The API can either the first package found (based on ordering of collections within a profile) or "merged" data.

We can't really do "local read first and remote fetch if needed" as we do with collection (i.e., preview), because remote doesn't have all the package metadata we need (e.g., targets, products), therefore the requested package must belong to a collection.

Reply:
good point!